### PR TITLE
Move UnsafeUtil to it's own module

### PR DESCRIPTION
**Pull request by chrisvest**
This way we end up with just one place where we play with fire, and just one place where we have to cope with compatibility with CPU architectures, operating systems and JREs.

Issued against master so both 2.2 and 2.3+ get on common ground with this.

### CODE REVIEW

File: community/unsafe/src/main/java/org/neo4j/unsafe/impl/internal/dragons/UnsafeUtil.java [hunk 1]
```diff
@@ -17,7 +17,7 @@
  * You should have received a copy of the GNU General Public License
  * along with this program.  If not, see <http://www.gnu.org/licenses/>.
  */
-package org.neo4j.io.pagecache.impl.muninn;
+package org.neo4j.unsafe.impl.internal.dragons;
```
Comment (davidegrohmann) [inline community/unsafe/src/main/java/org/neo4j/unsafe/impl/internal/dragons/UnsafeUtil.java]: nice

File: community/unsafe/pom.xml [hunk 2]
```diff
@@ -0,0 +1,63 @@
...
+  <version>2.2-SNAPSHOT</version>
+
+  <packaging>jar</packaging>
+  <name>Neo4j - Unsafe Access</name>
+  <description>Internal component that provides access to unsafe memory
+    intrinsics</description>
+  <url>http://components.neo4j.org/${project.artifactId}/${project.version}</url>
+
+  <scm>
+    <url>https://github.com/neo4j/neo4j/tree/master/community/primitive-collections</url>
```
Comment (chrisvest) [inline community/unsafe/pom.xml]: Oops.

File: community/unsafe/src/main/java/org/neo4j/unsafe/impl/internal/dragons/UnsafeUtil.java (hunk 1 above)
Comment (tinwelint) [inline community/unsafe/src/main/java/org/neo4j/unsafe/impl/internal/dragons/UnsafeUtil.java]: now that's a thing you don't wanna go ahead and use right off the bat. Good and proper name.

**Comment by tinwelint** [comment]
Great that you did pick up those other users of unsafe, OffHeapNumberArray and UnsafeTable. Looks good to me

**Comment by zhenlineo** [comment]
:+1:

**Pull request merged**